life_game/game.rb

```diff
@@ -0,0 +1,151 @@
+  end
+
+  def self.r_board
+    top_bottom = Array.new(SIZE*10,0)
+    zeros = Array.new(10,0)
+
+    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
+    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
+    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
+    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
+    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
+
+    top_r = r1+r2+r3+r4+r5
+    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
+
+    board = top_bottom+top_r+bottom_r*5+top_bottom
+    start_board = Array.new(SIZE){Array.new}
+    i = 0
+    start_board.each do |row|
+      board[i..i+SIZE-1].each do |cell|
+        row<<cell
+      end
+      i = i+SIZE
+    end
+    start_board
+  end
+
+  def self.neighbours(i,j)
+    size = SIZE-1
+    if i != 0 && j != 0 && i != size && j != size
+      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
+    elsif i == 0 && j != 0  && j != size
+      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
+    elsif i == size && j != 0 && j != size
+      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
+    elsif j == 0 && i != 0 && i != size
+      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
+    elsif j == size && i != 0 && i != size
+      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
+    elsif i == 0 && j == 0
+      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
+    elsif i == size && j == size
+      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
+    elsif i == 0 && j == size
+      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
+    else
+      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
+    end
+  end
+
+  def self.all_sum_neigbours
+    all_sum_neigbours = []
+    i = 0
+    @board.each do |row|
+      j= 0
+      row.each do |cell|
+        sum_neigbours = {}
+        sum_neigbours[cell] = neighbours(i,j).sum
+        all_sum_neigbours << sum_neigbours
+        j=j+1
+      end
+      i=i+1
+    end
+    all_sum_neigbours
+  end
+
+  def self.decision
+    decision = []
+    all_sum_neigbours.each do |hash|
+      hash.each do |key, value|
+        if key == 0
+          if value == 3
+            decision<<1
+          else
+            decision<<0
+          end
+        else
+          if value == 2 || value == 3
+            decision<<1
+          else
+            decision<<0
+          end
+        end
+      end
+    end
+    decision
+  end
+
+  def self.new_board
+    new_board = Array.new(SIZE){Array.new}
+    i = 0
+    new_board.each do |row|
+      decision[i..i+SIZE-1].each do |cell|
+        row<<cell
+      end
+      i = i+SIZE
+    end
+    @board = new_board
+  end
+
+end
```

Comment:
tutaj wkleję output z rubocopa:

```
$ rubocop game.rb
Inspecting 1 file
W

Offenses:

game.rb:1:1: C: Class has too many lines. [134/100]
class Game ...
^^^^^^^^^^
game.rb:1:1: C: Missing top-level class documentation comment.
class Game
^^^^^
game.rb:2:1: C: Use 2 (not 0) spaces for indentation.
attr_accessor :board

game.rb:5:3: C: Inconsistent indentation detected.
  def initialize(board) ...
  ^^^^^^^^^^^^^^^^^^^^^
game.rb:9:3: C: Inconsistent indentation detected.
  def self.input_board(input) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:10:5: C: Use the return of the conditional for variable assignment and comparison.
    if input == nil ...
    ^^^^^^^^^^^^^^^
game.rb:10:14: C: Prefer the use of the nil? predicate.
    if input == nil
             ^^
game.rb:17:1: C: Extra blank line detected.
game.rb:18:3: C: Use empty lines between method definitions.
  def self.empty_board
  ^^^
game.rb:18:3: C: Inconsistent indentation detected.
  def self.empty_board ...
  ^^^^^^^^^^^^^^^^^^^^
game.rb:19:20: C: Space missing to the left of {.
    Array.new(SIZE){Array.new(SIZE,0)}
                   ^
game.rb:19:21: C: Space missing inside {.
    Array.new(SIZE){Array.new(SIZE,0)}
                    ^
game.rb:19:35: C: Space missing after comma.
    Array.new(SIZE){Array.new(SIZE,0)}
                                  ^
game.rb:19:38: C: Space missing inside }.
    Array.new(SIZE){Array.new(SIZE,0)}
                                     ^
game.rb:22:3: C: Inconsistent indentation detected.
  def self.params_array(input) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:22:3: C: Method has too many lines. [15/10]
  def self.params_array(input) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:23:6: C: Surrounding space missing for operator =.
    i=0
     ^
game.rb:24:17: C: Surrounding space missing for operator =.
    params_array=[]
                ^
game.rb:26:7: C: Use the return of the conditional for variable assignment and comparison.
      if cell=="1" ...
      ^^^^^^^^^^^^
game.rb:26:14: C: Surrounding space missing for operator ==.
      if cell=="1"
             ^^
game.rb:26:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      if cell=="1"
               ^^^
game.rb:27:13: C: Surrounding space missing for operator =.
        cell=1
            ^
game.rb:28:20: C: Surrounding space missing for operator +.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                   ^
game.rb:28:23: C: Surrounding space missing for operator ==.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                      ^^
game.rb:28:25: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                        ^^^^^
game.rb:28:30: C: Surrounding space missing for operator ||.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                             ^^
game.rb:28:39: C: Surrounding space missing for operator +.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                                      ^
game.rb:28:42: C: Surrounding space missing for operator ==.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                                         ^^
game.rb:28:42: C: Prefer the use of the nil? predicate.
      elsif input[i+1]=="OFF"||input[i+1]==nil
                                         ^^
game.rb:29:7: C: Use 2 (not 4) spaces for indentation.
          cell=0
      ^^^^
game.rb:29:15: C: Surrounding space missing for operator =.
          cell=0
              ^
game.rb:31:7: C: Use 2 (not 4) spaces for indentation.
          cell=cell
      ^^^^
game.rb:31:15: C: Surrounding space missing for operator =.
          cell=cell
              ^
game.rb:33:19: C: Surrounding space missing for operator <<.
      params_array<<cell
                  ^^
game.rb:34:7: C: Use self-assignment shorthand +=.
      i=i+1
      ^^^^^
game.rb:34:8: C: Surrounding space missing for operator =.
      i=i+1
       ^
game.rb:34:10: C: Surrounding space missing for operator +.
      i=i+1
         ^
game.rb:36:25: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    params_array.delete("OFF")
                        ^^^^^
game.rb:40:3: C: Inconsistent indentation detected.
  def self.params_board(input) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:41:35: C: Space missing to the left of {.
    params_board = Array.new(SIZE){Array.new}
                                  ^
game.rb:41:36: C: Space missing inside {.
    params_board = Array.new(SIZE){Array.new}
                                   ^
game.rb:41:36: C: Use array literal [] instead of Array.new.
    params_board = Array.new(SIZE){Array.new}
                                   ^^^^^^^^^
game.rb:41:45: C: Space missing inside }.
    params_board = Array.new(SIZE){Array.new}
                                            ^
game.rb:45:24: C: Surrounding space missing for operator +.
      params_array[i..i+SIZE-1].each do |cell|
                       ^
game.rb:45:29: C: Surrounding space missing for operator -.
      params_array[i..i+SIZE-1].each do |cell|
                            ^
game.rb:46:12: C: Surrounding space missing for operator <<.
        row<<cell
           ^^
game.rb:48:7: C: Use self-assignment shorthand +=.
      i = i+SIZE
      ^^^^^^^^^^
game.rb:48:12: C: Surrounding space missing for operator +.
      i = i+SIZE
           ^
game.rb:53:3: C: Inconsistent indentation detected.
  def self.r_board ...
  ^^^^^^^^^^^^^^^^
game.rb:53:3: C: Assignment Branch Condition size for r_board is too high. [57.49/15]
  def self.r_board
  ^^^
game.rb:53:3: C: Method has too many lines. [19/10]
  def self.r_board ...
  ^^^^^^^^^^^^^^^^
game.rb:54:32: C: Surrounding space missing for operator *.
    top_bottom = Array.new(SIZE*10,0)
                               ^
game.rb:54:35: C: Space missing after comma.
    top_bottom = Array.new(SIZE*10,0)
                                  ^
game.rb:55:25: C: Space missing after comma.
    zeros = Array.new(10,0)
                        ^
game.rb:57:15: C: Surrounding space missing for operator +.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
              ^
game.rb:57:27: C: Space missing after comma.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                          ^
game.rb:57:30: C: Surrounding space missing for operator +.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                             ^
game.rb:57:42: C: Space missing after comma.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                                         ^
game.rb:57:45: C: Surrounding space missing for operator +.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                                            ^
game.rb:57:57: C: Space missing after comma.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                                                        ^
game.rb:57:60: C: Surrounding space missing for operator +.
    r1 = zeros+Array.new(3,1)+Array.new(4,0)+Array.new(3,1)+zeros
                                                           ^
game.rb:58:15: C: Surrounding space missing for operator +.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
              ^
game.rb:58:27: C: Space missing after comma.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                          ^
game.rb:58:30: C: Surrounding space missing for operator +.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                             ^
game.rb:58:42: C: Space missing after comma.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                                         ^
game.rb:58:45: C: Surrounding space missing for operator +.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                                            ^
game.rb:58:57: C: Space missing after comma.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                                                        ^
game.rb:58:60: C: Surrounding space missing for operator +.
    r2 = zeros+Array.new(3,1)+Array.new(2,0)+Array.new(5,1)+zeros
                                                           ^
game.rb:59:15: C: Surrounding space missing for operator +.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
              ^
game.rb:59:27: C: Space missing after comma.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                          ^
game.rb:59:30: C: Surrounding space missing for operator +.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                             ^
game.rb:59:42: C: Space missing after comma.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                                         ^
game.rb:59:45: C: Surrounding space missing for operator +.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                                            ^
game.rb:59:57: C: Space missing after comma.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                                                        ^
game.rb:59:60: C: Surrounding space missing for operator +.
    r3 = zeros+Array.new(3,1)+Array.new(1,0)+Array.new(6,1)+zeros
                                                           ^
game.rb:60:15: C: Surrounding space missing for operator +.
    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
              ^
game.rb:60:27: C: Space missing after comma.
    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
                          ^
game.rb:60:30: C: Surrounding space missing for operator +.
    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
                             ^
game.rb:60:42: C: Space missing after comma.
    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
                                         ^
game.rb:60:45: C: Surrounding space missing for operator +.
    r4 = zeros+Array.new(6,1)+Array.new(4,0)+zeros
                                            ^
game.rb:61:15: C: Surrounding space missing for operator +.
    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
              ^
game.rb:61:27: C: Space missing after comma.
    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
                          ^
game.rb:61:30: C: Surrounding space missing for operator +.
    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
                             ^
game.rb:61:42: C: Space missing after comma.
    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
                                         ^
game.rb:61:45: C: Surrounding space missing for operator +.
    r5 = zeros+Array.new(4,1)+Array.new(6,0)+zeros
                                            ^
game.rb:63:15: C: Surrounding space missing for operator +.
    top_r = r1+r2+r3+r4+r5
              ^
game.rb:63:18: C: Surrounding space missing for operator +.
    top_r = r1+r2+r3+r4+r5
                 ^
game.rb:63:21: C: Surrounding space missing for operator +.
    top_r = r1+r2+r3+r4+r5
                    ^
game.rb:63:24: C: Surrounding space missing for operator +.
    top_r = r1+r2+r3+r4+r5
                       ^
game.rb:64:21: C: Surrounding space missing for operator +.
    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
                    ^
game.rb:64:33: C: Space missing after comma.
    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
                                ^
game.rb:64:36: C: Surrounding space missing for operator +.
    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
                                   ^
game.rb:64:48: C: Space missing after comma.
    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
                                               ^
game.rb:64:51: C: Surrounding space missing for operator +.
    bottom_r = zeros+Array.new(3,1)+Array.new(7,0)+zeros
                                                  ^
game.rb:66:23: C: Surrounding space missing for operator +.
    board = top_bottom+top_r+bottom_r*5+top_bottom
                      ^
game.rb:66:29: C: Surrounding space missing for operator +.
    board = top_bottom+top_r+bottom_r*5+top_bottom
                            ^
game.rb:66:38: C: Surrounding space missing for operator *.
    board = top_bottom+top_r+bottom_r*5+top_bottom
                                     ^
game.rb:66:40: C: Surrounding space missing for operator +.
    board = top_bottom+top_r+bottom_r*5+top_bottom
                                       ^
game.rb:67:34: C: Space missing to the left of {.
    start_board = Array.new(SIZE){Array.new}
                                 ^
game.rb:67:35: C: Space missing inside {.
    start_board = Array.new(SIZE){Array.new}
                                  ^
game.rb:67:35: C: Use array literal [] instead of Array.new.
    start_board = Array.new(SIZE){Array.new}
                                  ^^^^^^^^^
game.rb:67:44: C: Space missing inside }.
    start_board = Array.new(SIZE){Array.new}
                                           ^
game.rb:70:17: C: Surrounding space missing for operator +.
      board[i..i+SIZE-1].each do |cell|
                ^
game.rb:70:22: C: Surrounding space missing for operator -.
      board[i..i+SIZE-1].each do |cell|
                     ^
game.rb:71:12: C: Surrounding space missing for operator <<.
        row<<cell
           ^^
game.rb:73:7: C: Use self-assignment shorthand +=.
      i = i+SIZE
      ^^^^^^^^^^
game.rb:73:12: C: Surrounding space missing for operator +.
      i = i+SIZE
           ^
game.rb:78:3: C: Inconsistent indentation detected.
  def self.neighbours(i,j) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:78:3: C: Assignment Branch Condition size for neighbours is too high. [160.8/15]
  def self.neighbours(i,j)
  ^^^
game.rb:78:3: C: Cyclomatic complexity for neighbours is too high. [23/6]
  def self.neighbours(i,j)
  ^^^
game.rb:78:3: C: Method has too many lines. [20/10]
  def self.neighbours(i,j) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:78:3: C: Perceived complexity for neighbours is too high. [24/7]
  def self.neighbours(i,j)
  ^^^
game.rb:78:24: C: Space missing after comma.
  def self.neighbours(i,j)
                       ^
game.rb:79:16: C: Surrounding space missing for operator -.
    size = SIZE-1
               ^
game.rb:81:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
      ^^^^^^^^^^
game.rb:81:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                            ^
game.rb:81:34: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                 ^
game.rb:81:37: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                    ^
game.rb:81:46: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                             ^
game.rb:81:52: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                   ^
game.rb:81:61: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                            ^
game.rb:81:66: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                 ^
game.rb:81:69: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                    ^
game.rb:81:81: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                ^
game.rb:81:81: C: Line is too long. [148/80]
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:81:84: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                   ^
game.rb:81:96: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                               ^
game.rb:81:99: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                  ^
game.rb:81:108: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                           ^
game.rb:81:113: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                ^
game.rb:81:116: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                   ^
game.rb:81:125: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                            ^
game.rb:81:131: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                                  ^
game.rb:81:140: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                                           ^
game.rb:81:145: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                                                                                ^
game.rb:82:11: C: Use i.zero? instead of i == 0.
    elsif i == 0 && j != 0  && j != size
          ^^^^^^
game.rb:82:29: C: Operator && should be surrounded by a single space.
    elsif i == 0 && j != 0  && j != size
                            ^^
game.rb:83:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
      ^^^^^^^^^^
game.rb:83:32: C: Surrounding space missing for operator -.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                               ^
game.rb:83:35: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                  ^
game.rb:83:47: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                              ^
game.rb:83:50: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                 ^
game.rb:83:59: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                          ^
game.rb:83:64: C: Surrounding space missing for operator -.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                               ^
game.rb:83:67: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                  ^
game.rb:83:76: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                           ^
game.rb:83:81: C: Line is too long. [99/80]
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                ^^^^^^^^^^^^^^^^^^^
game.rb:83:82: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                 ^
game.rb:83:91: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                          ^
game.rb:83:96: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i][j+1],@board[i+1][j-1],@board[i+1][j],@board[i+1][j+1]]
                                                                                               ^
game.rb:85:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
      ^^^^^^^^^^
game.rb:85:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                            ^
game.rb:85:34: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                 ^
game.rb:85:37: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                    ^
game.rb:85:46: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                             ^
game.rb:85:52: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                   ^
game.rb:85:61: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                            ^
game.rb:85:66: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                 ^
game.rb:85:69: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                    ^
game.rb:85:81: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                                ^
game.rb:85:81: C: Line is too long. [99/80]
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                                ^^^^^^^^^^^^^^^^^^^
game.rb:85:84: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                                   ^
game.rb:85:96: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i-1][j+1],@board[i][j-1],@board[i][j+1]]
                                                                                               ^
game.rb:86:11: C: Use j.zero? instead of j == 0.
    elsif j == 0 && i != 0 && i != size
          ^^^^^^
game.rb:87:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
      ^^^^^^^^^^
game.rb:87:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                            ^
game.rb:87:35: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                  ^
game.rb:87:44: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                           ^
game.rb:87:49: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                ^
game.rb:87:52: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                   ^
game.rb:87:64: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                               ^
game.rb:87:67: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                  ^
game.rb:87:76: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                           ^
game.rb:87:81: C: Line is too long. [99/80]
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                                ^^^^^^^^^^^^^^^^^^^
game.rb:87:82: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                                 ^
game.rb:87:91: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                                          ^
game.rb:87:96: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                                                               ^
game.rb:89:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
      ^^^^^^^^^^
game.rb:89:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                            ^
game.rb:89:34: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                 ^
game.rb:89:37: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                    ^
game.rb:89:46: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                             ^
game.rb:89:52: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                   ^
game.rb:89:64: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                               ^
game.rb:89:67: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                  ^
game.rb:89:76: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                           ^
game.rb:89:81: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                                ^
game.rb:89:81: C: Line is too long. [99/80]
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                                ^^^^^^^^^^^^^^^^^^^
game.rb:89:84: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                                   ^
game.rb:89:93: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                                                            ^
game.rb:90:11: C: Use i.zero? instead of i == 0.
    elsif i == 0 && j == 0
          ^^^^^^
game.rb:90:21: C: Use j.zero? instead of j == 0.
    elsif i == 0 && j == 0
                    ^^^^^^
game.rb:91:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
      ^^^^^^^^^^
game.rb:91:32: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                               ^
game.rb:91:35: C: Space missing after comma.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                  ^
game.rb:91:44: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                           ^
game.rb:91:50: C: Space missing after comma.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                 ^
game.rb:91:59: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                          ^
game.rb:91:64: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j+1],@board[i+1][j],@board[i+1][j+1]]
                                                               ^
game.rb:93:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
      ^^^^^^^^^^
game.rb:93:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                            ^
game.rb:93:34: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                                 ^
game.rb:93:37: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                                    ^
game.rb:93:46: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                                             ^
game.rb:93:52: C: Space missing after comma.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                                                   ^
game.rb:93:64: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j-1],@board[i-1][j],@board[i][j-1]]
                                                               ^
game.rb:94:11: C: Use i.zero? instead of i == 0.
    elsif i == 0 && j == size
          ^^^^^^
game.rb:95:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
      ^^^^^^^^^^
game.rb:95:32: C: Surrounding space missing for operator -.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                               ^
game.rb:95:35: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                  ^
game.rb:95:44: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                           ^
game.rb:95:49: C: Surrounding space missing for operator -.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                ^
game.rb:95:52: C: Space missing after comma.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                   ^
game.rb:95:61: C: Surrounding space missing for operator +.
      neighbours = [@board[i][j-1],@board[i+1][j-1],@board[i+1][j]]
                                                            ^
game.rb:97:7: W: Useless assignment to variable - neighbours.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
      ^^^^^^^^^^
game.rb:97:29: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                            ^
game.rb:97:35: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                                  ^
game.rb:97:44: C: Surrounding space missing for operator -.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                                           ^
game.rb:97:49: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                                                ^
game.rb:97:52: C: Space missing after comma.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                                                   ^
game.rb:97:64: C: Surrounding space missing for operator +.
      neighbours = [@board[i-1][j],@board[i-1][j+1],@board[i][j+1]]
                                                               ^
game.rb:101:3: C: Inconsistent indentation detected.
  def self.all_sum_neigbours ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:101:3: C: Method has too many lines. [13/10]
  def self.all_sum_neigbours ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:105:8: C: Surrounding space missing for operator =.
      j= 0
       ^
game.rb:108:43: C: Space missing after comma.
        sum_neigbours[cell] = neighbours(i,j).sum
                                          ^
game.rb:110:9: C: Use self-assignment shorthand +=.
        j=j+1
        ^^^^^
game.rb:110:10: C: Surrounding space missing for operator =.
        j=j+1
         ^
game.rb:110:12: C: Surrounding space missing for operator +.
        j=j+1
           ^
game.rb:112:7: C: Use self-assignment shorthand +=.
      i=i+1
      ^^^^^
game.rb:112:8: C: Surrounding space missing for operator =.
      i=i+1
       ^
game.rb:112:10: C: Surrounding space missing for operator +.
      i=i+1
         ^
game.rb:117:3: C: Inconsistent indentation detected.
  def self.decision ...
  ^^^^^^^^^^^^^^^^^
game.rb:117:3: C: Method has too many lines. [19/10]
  def self.decision ...
  ^^^^^^^^^^^^^^^^^
game.rb:117:3: C: Perceived complexity for decision is too high. [8/7]
  def self.decision
  ^^^
game.rb:121:12: C: Use key.zero? instead of key == 0.
        if key == 0
           ^^^^^^^^
game.rb:122:11: C: Use the return of the conditional for variable assignment and comparison.
          if value == 3 ...
          ^^^^^^^^^^^^^
game.rb:123:21: C: Surrounding space missing for operator <<.
            decision<<1
                    ^^
game.rb:125:21: C: Surrounding space missing for operator <<.
            decision<<0
                    ^^
game.rb:128:11: C: Use the return of the conditional for variable assignment and comparison.
          if value == 2 || value == 3 ...
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:128:11: C: Convert if nested inside else to elsif.
          if value == 2 || value == 3
          ^^
game.rb:128:11: C: Avoid comparing a variable with multiple items in a conditional, use Array#include? instead.
          if value == 2 || value == 3 ...
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
game.rb:129:21: C: Surrounding space missing for operator <<.
            decision<<1
                    ^^
game.rb:131:21: C: Surrounding space missing for operator <<.
            decision<<0
                    ^^
game.rb:139:3: C: Inconsistent indentation detected.
  def self.new_board ...
  ^^^^^^^^^^^^^^^^^^
game.rb:140:32: C: Space missing to the left of {.
    new_board = Array.new(SIZE){Array.new}
                               ^
game.rb:140:33: C: Space missing inside {.
    new_board = Array.new(SIZE){Array.new}
                                ^
game.rb:140:33: C: Use array literal [] instead of Array.new.
    new_board = Array.new(SIZE){Array.new}
                                ^^^^^^^^^
game.rb:140:42: C: Space missing inside }.
    new_board = Array.new(SIZE){Array.new}
                                         ^
game.rb:143:20: C: Surrounding space missing for operator +.
      decision[i..i+SIZE-1].each do |cell|
                   ^
game.rb:143:25: C: Surrounding space missing for operator -.
      decision[i..i+SIZE-1].each do |cell|
                        ^
game.rb:144:12: C: Surrounding space missing for operator <<.
        row<<cell
           ^^
game.rb:146:7: C: Use self-assignment shorthand +=.
      i = i+SIZE
      ^^^^^^^^^^
game.rb:146:12: C: Surrounding space missing for operator +.
      i = i+SIZE
           ^
game.rb:150:1: C: Extra empty line detected at class body end.
```